examples/ensemble/plot_gradient_boosting_categorical.py

```diff
@@ -0,0 +1,211 @@
+#
+# The main difference between this pipeline and the previous one is that in
+# this one, we let the :class:`~ensemble.HistGradientBoostingRegressor` know
+# which features are categorical.
+
+# The ordinal encoder will first output the categorical features, and then the
+# continuous (passed-through) features
+categorical_mask = ([True] * n_categorical_features +
+                    [False] * n_numerical_features)
+hist_native = make_pipeline(
+    ordinal_encoder,
+    HistGradientBoostingRegressor(random_state=42,
+                                  categorical_features=categorical_mask)
+)
+
+
+# %%
+# Model comparison
+# ----------------
+# Finally, we evaluate the models using cross validation. Here we compare the
+# models performance in terms of
+# :func:`~metrics.mean_absolute_percentage_error` and fit times.
+
+from sklearn.model_selection import cross_validate
+import matplotlib.pyplot as plt
+
+scoring = "neg_mean_absolute_percentage_error"
+dropped_result = cross_validate(hist_dropped, X, y, cv=3, scoring=scoring)
+one_hot_result = cross_validate(hist_one_hot, X, y, cv=3, scoring=scoring)
+ordinal_result = cross_validate(hist_ordinal, X, y, cv=3, scoring=scoring)
+native_result = cross_validate(hist_native, X, y, cv=3, scoring=scoring)
+
+
+def plot_results(figure_title):
+    fig, (ax1, ax2) = plt.subplots(1, 2, figsize=(12, 8))
+
+    plot_info = [('fit_time', 'Fit times (s)', ax1, None),
+                 ('test_score', 'Mean Absolute Percentage Error', ax2,
+                  (0, 0.20))]
+
+    x, width = np.arange(4), 0.9
+    for key, title, ax, y_limit in plot_info:
+        items = [dropped_result[key], one_hot_result[key], ordinal_result[key],
+                 native_result[key]]
+        ax.bar(x, [np.mean(np.abs(item)) for item in items],
+               width, yerr=[np.std(item) for item in items],
+               color=['C0', 'C1', 'C2', 'C3'])
+        ax.set(xlabel='Model', title=title, xticks=x,
+               xticklabels=["Dropped", "One Hot", "Ordinal", "Native"],
+               ylim=y_limit)
+    fig.suptitle(figure_title)
+
+
+plot_results("Gradient Boosting on Adult Census")
+
+# %%
+# We see that the model with one-hot-encoded data is by far the slowest. This
+# is to be expected, since one-hot-encoding creates one additional feature per
+# category value (for each categorical feature), and thus more split points
+# need to be considered during fitting. The native handling of categorical
+# features is slightly slower than treating categories as ordered quantities
+# ('Ordinal'), since native handling requires :ref:`sorting categories
+# <categorical_support_gbdt>`.
+#
+# In terms of prediction performance, dropping the categorical features leads
+# to poorer performance. The three models that use categorical features have
+# comparable error rates, with a slight edge for the native handling.
+#
+# In general, one can expect poorer predictions from one-hot-encoded data,
```

Comment:
Should we move this paragraph into the next section?

Reply:
Let me do this.